### PR TITLE
Clean up document TODOs

### LINKED ISSUE #260: track removal of documentation TODOs

### PR DESCRIPTION
**Pull request by calebofearth**
Add new content to resolve TODO items in Caliptra Integration Specification and Caliptra Hardware Specification.

Resolves #260

### CODE REVIEW

**Comment by calebofearth** [comment]
On removal of the Keygen TVLA section: I filed #346 to track this for a future addition, but we should proceed without it for now. 

I consider this PR ready to merge.

@andreslagarcavilla Once this PR is approved, can you please merge it using the squash strategy? No need to preserve the history on this one since I will just delete the branch later.

File: docs/CaliptraHardwareSpecification.md [hunk 1]
```diff
@@ -396,38 +397,8 @@ The UART block architecture inputs and outputs are described in the following ta
...
-| KEY_MANIFEST_PK_HASH      | 0x300302b0        | |
-| KEY_MANIFEST_PK_HASH_MASK | 0x30030370        | |
-| KEY_MANIFEST_SVN          | 0x30030374        | |
-| BOOT_LOADER_SVN           | 0x30030384        | |
-| RUNTIME_SVN               | 0x30030388        | |
-| ANTI_ROLLBACK_DISABLE     | 0x3003038c        | |
-| IEEE_IDEVID_CERT_CHAIN    | 0x30030390        | |
-| FUSE_DONE                 | 0x300303f0        | |
+For more information on the mailbox protocol, see [Mailbox](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#mailbox) in the Caliptra Integration Specification. Registers used to interact with the Mailbox are defined [here](https://chipsalliance.github.io/caliptra-rtl/main/internal-regs/?p=clp.mbox_csr).
+
```
Comment (steph-morton) [inline docs/CaliptraHardwareSpecification.md:401]: Recommend update last sentence to: Registers that interact with the Mailbox are defined in [mbox_csr](https://chipsalliance.github.io/caliptra-rtl/main/internal-regs/?p=clp.mbox_csr).

File: docs/CaliptraHardwareSpecification.md [hunk 2]
```diff
@@ -143,7 +143,8 @@ Software and local interrupts are not implemented in the first generation of Cal
 
 ##### Non-maskable interrupts
 
-<TODO> 0p8 describe a register bank that may be used to dynamically configure the NMI reset vector. (i.e., where the PC resets to).
+Caliptra's RISC-V processor has access to an internal register that allows configuration of the Non-Maskable Interrupt vector. When an NMI occurs, the program counter will jump to the address indicated by the contents of this register.
```
[the remark this answers is not in the document]
Comment (calebofearth) [reply docs/CaliptraHardwareSpecification.md]: Good catch - fixed.

File: docs/CaliptraHardwareSpecification.md (hunk 1 above)
Comment (calebofearth) [reply docs/CaliptraHardwareSpecification.md:401]: The registers are part of the mailbox moreso than interacting with it -- how about `For more information about Mailbox registers, see [mbox_csr].`

File: docs/CaliptraHardwareSpecification.md [hunk 3]
```diff
@@ -143,7 +143,8 @@ Software and local interrupts are not implemented in the first generation of Cal
...
+Refer to [NMI Vector](https://chipsalliance.github.io/caliptra-rtl/main/internal-regs/?p=clp.soc_ifc_reg.internal_nmi_vector) for official documentation.
```
[the remark this answers is not in the document]
Comment (calebofearth) [reply docs/CaliptraHardwareSpecification.md]: Done

File: docs/CaliptraHardwareSpecification.md (hunk 1 above)
Comment (calebofearth) [reply docs/CaliptraHardwareSpecification.md:401]: Pushed an update, LMK what you think.

**Review by steph-morton — APPROVED** [review]

File: docs/CaliptraHardwareSpecification.md (hunk 1 above)
Comment (steph-morton) [reply docs/CaliptraHardwareSpecification.md:401]: Updates look good, thanks!

**Review by andreslagarcavilla — APPROVED** [review]

**Pull request merged**